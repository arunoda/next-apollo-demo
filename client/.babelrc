{
  "presets": ["next/babel"],
  "env": {
    "production": {
      "plugins": [
        ["react-remove-properties", {"properties": ["data-testid"]}]
      ]
    }
  }
}
