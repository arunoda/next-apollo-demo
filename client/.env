NEXT_PUBLIC_GRAPHQL_SERVER=http://localhost:5000/graphql