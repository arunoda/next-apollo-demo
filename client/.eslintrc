{
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "node": true
  },
  "extends": [
    "airbnb",
    "airbnb/hooks",
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:jsx-a11y/recommended",
    // "plugin:react-hooks/recommended",
    // always put prettier at last
    "prettier"
  ],
  "globals": {
    "Atomics": "readonly",
    "SharedArrayBuffer": "readonly"
  },
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true // enable linting for jsx files
    },
    "ecmaVersion": 11,
    "sourceType": "module"
  },
  "settings": {
    "react": {
      "version": "detect"
    }
    // "import/resolver": {
    //   "alias": { "map": [["src", "./src"]], "extensions": [".jsx", ".js"] }
    // }
  },
  "plugins": ["react", "react-hooks"],
  "rules": {
    "react/jsx-props-no-spreading": 0,
    "react/forbid-prop-types": 0
  }
}
