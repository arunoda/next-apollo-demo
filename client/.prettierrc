{
  "arrowParens": "avoid",
  "semi": true,
  "trailingComma": "all",
  "singleQuote": true,
  "printWidth": 120
}