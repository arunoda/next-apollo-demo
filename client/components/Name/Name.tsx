import { useQuery } from '@apollo/react-hooks';
import gql from 'graphql-tag'

interface NameData {
  name: string;
}

const Name = (): JSX.Element => {
  const { loading, data, error } = useQuery<NameData>(GET_NAME, {
    variables: { language: 'english' },
  });

  return (
    <span>
      {loading || error ? '..' : data?.name}
    </span>
  );
}

const GET_NAME = gql`
  query name {
    name
  }
`

export default Name
