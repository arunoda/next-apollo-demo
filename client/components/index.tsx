export { default as Name } from './Name'
export { default as UserTile } from './UserTile'