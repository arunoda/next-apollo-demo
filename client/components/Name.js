import React from 'react';
import { graphql } from '@apollo/client/react/hoc';
import { gql } from '@apollo/client'

const Component = ({ data }) => (
  <span>
    {data.loading? '..' : data.name}
  </span>
)

const query = gql`
  query name {
    name
  }
`

export default graphql(query)(Component)
