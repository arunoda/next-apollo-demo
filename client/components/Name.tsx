import { graphql } from 'react-apollo'
import {gql} from '@apollo/client'

const Component = ({ data }) => (
  <span>
    {data.loading? '..' : data.name}
  </span>
)

const query = gql`
  query name {
    name
  }
`

export default graphql(query)(Component)
