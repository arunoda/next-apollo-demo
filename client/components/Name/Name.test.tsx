import React from 'react';
import { useQuery } from "@apollo/client";
import { render } from '@testing-library/react';
import { Name } from './Name';

jest.mock('@apollo/client');

const mockUseQuery = useQuery as jest.MockedFunction<any>;

describe('Name', () => {

  beforeEach(() => {
    mockUseQuery.mockReturnValue({
      data: {
        name: "Name",
      }
    })
  })

  it('should match snapshot', () => {
    const { container } = render(<Name />);
    expect(container).toMatchSnapshot();
  })
})