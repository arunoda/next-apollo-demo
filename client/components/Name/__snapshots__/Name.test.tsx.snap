// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name should match snapshot 1`] = `
<div>
  <span>
    Name
  </span>
</div>
`;
